lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -192,6 +190,22 @@ def add_billing_address(post, payment, options)
         post[:billingAddress] = billing_address
       end
 
+      def name_from_payment(billing_address, payment)
+        return unless payment.first_name && payment.last_name
```

Comment:
The other reason I think we might want to use `respond_to?` here is that the previous logic would in theory allow for us to send just a first name or just a last name, if by chance only one was included.

Reply:
heya @jcreiff a little confused about this but from what im gathering, we're trying to not send any name fields if both do not exist correct? In that case, I always thought respond_to? was to see if the attribute/class method exists (even if its nil). I did some testing for this and if i replace that line with something like return unless payment.respond_to?(:first_name) && payment.respond_to?(:last_name) and set the credit_card.last_name for example to nil, it will still proceed and try to map the both names. If we're trying to skip this method if only one name was sent, wouldn't the return unless payment.first_name && payment.last_name cover that? 🧐

Reply:
> we're trying to not send any name fields if both do not exist correct?

The way it is originally written on lines 175-76, it doesn't look like that is an expectation. (although...should it be? 🤷🏻 )

What I was thinking was this:
- remove the `if payment.is_a?(CreditCard)` check from 175 so that we can add names for non-CC payment methods
- add the `respond_to` guard clause at the top of `name_from_payment` so that we don't try to call those methods on something like a third party token, for example
- then if you keep the `if payment.first_name` and `if payment.last_name` nil checks at the ends of lines 196-97, you will still skip adding empty fields

Reply:
ahh gotcha, that makes more sense! so in theory, we will still send either the first_name or last_name fields even if only one of them exists

Reply:
with the return unless payment.first_name && payment.last_name at the top originally, it would skip the entire method and return out if both name fields do not exist